bayesian/utils.py

```diff
@@ -329,8 +329,118 @@ def search_packages_from_graph(tokens):
     return {'result': pkg_list}
 
 
+class GraphAnalyses:
+    """It Queries GraphDB Based on vendor and returns json converted data."""
+
+    def __init__(self, ecosystem, package, version, vendor=None):
+        """For Flows related to Security Vendor Integrations."""
+        self.vendor = vendor
+        self.ecosystem = ecosystem
+        self.version = version
+        self.package = package
+
+    @staticmethod
+    def build_graphdb_query(vendor):
+        """Func returns GraphDB query based on vendor argument."""
+        get_cve_info = """
+            g.V().has('pecosystem', ecosystem).has('pname', name).has('version', version)
+            .as('version').in('has_version').dedup().as('package').select('version')
+            .coalesce(out('has_cve').as('cve').select('package','version','cve')
+            .by(valueMap()),select('package','version').by(valueMap()));
+            """
+
+        get_latest_non_cve_version = """
+            g.V().has('pecosystem', ecosystem).has('pname', name)
+            .has('version', version).in('has_version').out('has_version')
+            .not(out('has_cve')).values('version').dedup();
+            """
+        # vendor specific query
+        fetch_cve_info = {
+            'snyk': "query_to_fetch_snyk_fields",
+        }
+        return (
+            fetch_cve_info.get(vendor, get_cve_info),
+            get_latest_non_cve_version)
+
+    def main(self):
+        """Fetch analysis for given package+version from the graph database.
+
+        It finally builds and returns JSON Response. This is main Function.
+        """
+        if self.vendor is None:
+            # if vendor is None, resume Old Flow for fetching data from our GraphDB Node
+            return get_analyses_from_graph(self.ecosystem, self.package, self.version)
+        cve_info_query, latest_non_cve_version_query = self.build_graphdb_query(self.vendor)
+        payload = {
+            'gremlin': cve_info_query,
+            'bindings': {
+                'ecosystem': self.ecosystem,
+                'name': self.package,
+                'version': self.version
+            }
+        }
+        start = datetime.datetime.now()
+        try:
+            clubbed_data = []
+            graph_req = post(gremlin_url, data=json.dumps(payload))
+
+            if graph_req is None:
+                return None
+
+            resp = graph_req.json()
+            result_data = resp['result'].get('data')
+
+            if not (result_data and len(result_data) > 0):
+                # trigger unknown component flow in API for missing package
+                return None
+
+            clubbed_data.append({
+                "epv": result_data
+            })
+
+            if "cve" not in result_data[0]:
+                clubbed_data.append({
+                    "recommended_versions": []
+                })
+
+            if "latest_non_cve_version" in result_data[0]['package']:
```

Comment:
couple of points here
1. this all should go in the else case, i.e when cve information is present in the data
2. you wont get any field like latest_no_cve_version in this case. Change the logic here